cmd/agent/main.go

```diff
@@ -234,8 +263,28 @@ func (a *Agent) run(o *GrpcProxyAgentOptions) error {
 		return fmt.Errorf("failed to validate agent options with %v", err)
 	}
 
+	var k8sClient *kubernetes.Clientset
+	if o.kubeconfigPath != "" || o.enablePodCondition {
```

Comment:
If we do the previously mentioned validations, then the "|| o.enablePodCondition" check should be unnecessary. This is good as it will remove a confusing build config from flags edge case where kubeconfigPath == "". It also means we fail earlier on several invalid configurations, which is good.

Reply:
I agree that this logic is precarious, but how should we handle cases where the kubeconfig flag is intentionally left unset to instead use the in cluster config?